plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergGetTableStatsOperations.java

```diff
@@ -0,0 +1,115 @@
+import io.trino.spi.security.PrincipalType;
+import io.trino.testing.LocalQueryRunner;
+import org.intellij.lang.annotations.Language;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.util.Optional;
+
+import static com.google.inject.util.Modules.EMPTY_MODULE;
+import static io.trino.plugin.hive.metastore.file.FileHiveMetastore.createTestingFileHiveMetastore;
+import static io.trino.sql.planner.LogicalPlanner.Stage.OPTIMIZED_AND_VALIDATED;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static io.trino.transaction.TransactionBuilder.transaction;
+import static org.testng.Assert.assertEquals;
+
+@Test(singleThreaded = true) // counting metadata is a shared mutable state
+public class TestIcebergGetTableStatsOperations
+{
+    private final LocalQueryRunner queryRunner;
+    private final CountingAccessMetadata metadata;
+    private File baseDir;
+
+    public TestIcebergGetTableStatsOperations()
+    {
+        queryRunner = LocalQueryRunner.builder(testSessionBuilder().build())
+                .withMetadataProvider((featuresConfig, systemSecurityMetadata, transactionManager, globalFunctionCatalog, typeManager)
+                        -> new CountingAccessMetadata(new MetadataManager(featuresConfig, systemSecurityMetadata, transactionManager, globalFunctionCatalog, typeManager)))
+                .build();
+        queryRunner.installPlugin(new TpchPlugin());
+        queryRunner.createCatalog("tpch", "tpch", ImmutableMap.of());
+
+        baseDir = Files.createTempDir();
+        HiveMetastore metastore = createTestingFileHiveMetastore(baseDir);
+
+        queryRunner.createCatalog(
+                "iceberg",
+                new TestingIcebergConnectorFactory(Optional.of(metastore), Optional.empty(), EMPTY_MODULE),
+                ImmutableMap.of());
+
+        Database database = Database.builder()
+                .setDatabaseName("tiny")
+                .setOwnerName(Optional.of("public"))
+                .setOwnerType(Optional.of(PrincipalType.ROLE))
+                .build();
+        metastore.createDatabase(database);
+
+        metadata = (CountingAccessMetadata) queryRunner.getMetadata();
+    }
+
+    @BeforeClass
+    public void setUp()
+    {
+        queryRunner.execute("CREATE TABLE iceberg.tiny.orders AS SELECT * FROM tpch.tiny.orders");
+        queryRunner.execute("CREATE TABLE iceberg.tiny.lineitem AS SELECT * FROM tpch.tiny.lineitem");
+        queryRunner.execute("CREATE TABLE iceberg.tiny.customer AS SELECT * FROM tpch.tiny.customer");
+    }
+
+    @Test
+    public void testTwoWayJoin()
+    {
+        metadata.resetCounters();
+        planQuery("SELECT custkey, count(1) " +
+                "FROM iceberg.tiny.orders o, iceberg.tiny.lineitem l " +
+                "WHERE o.orderkey = l.orderkey " +
+                "GROUP BY 1 " +
+                "ORDER BY 2 DESC " +
+                "LIMIT 5");
```

Comment:
Does LIMIT matter? 
Does ORDER BY matter?
Does GROUP BY matter?

please have test with as simple query as possible (just a JOIN).
if additional clauses matter, they should be as additional tests

Reply:
(same in the other test class)